scripts/globals/job_utils/beastmaster.lua

```diff
@@ -0,0 +1,512 @@
+        end,
+    }
+
+    -- Now calculating the bonus based on gear.
+    switch (player:getEquipID(xi.slot.BODY)) : caseof {
+        [xi.item.BEAST_JACKCOAT] = function() -- beast jackcoat
+            -- This will remove Paralyze, Poison and Blind from the pet.
+            pet:delStatusEffect(xi.effect.PARALYSIS)
+            pet:delStatusEffect(xi.effect.POISON)
+            pet:delStatusEffect(xi.effect.BLINDNESS)
+        end,
+
+        [xi.item.BEAST_JACKCOAT_P1] = function() -- beast jackcoat +1
+            -- This will remove Paralyze, Poison, Blind, Weight, Slow and Silence from the pet.
+            pet:delStatusEffect(xi.effect.PARALYSIS)
+            pet:delStatusEffect(xi.effect.POISON)
+            pet:delStatusEffect(xi.effect.BLINDNESS)
+            pet:delStatusEffect(xi.effect.WEIGHT)
+            pet:delStatusEffect(xi.effect.SLOW)
+            pet:delStatusEffect(xi.effect.SILENCE)
+        end,
+
+        [xi.item.MONSTER_JACKCOAT] = function() -- monster jackcoat
+            -- This will remove Weight, Slow and Silence from the pet.
+            pet:delStatusEffect(xi.effect.WEIGHT)
+            pet:delStatusEffect(xi.effect.SLOW)
+            pet:delStatusEffect(xi.effect.SILENCE)
+        end,
+
+        [xi.item.MONSTER_JACKCOAT_P1] = function() -- monster jackcoat +1
+            -- This will remove Paralyze, Poison, Blind, Weight, Slow and Silence from the pet.
+            pet:delStatusEffect(xi.effect.PARALYSIS)
+            pet:delStatusEffect(xi.effect.POISON)
+            pet:delStatusEffect(xi.effect.BLINDNESS)
+            pet:delStatusEffect(xi.effect.WEIGHT)
+            pet:delStatusEffect(xi.effect.SLOW)
+            pet:delStatusEffect(xi.effect.SILENCE)
+        end,
+    }
+
+    -- Adding bonus to the total to heal.
+
+    if
+        rewardHealingMod ~= nil and
+        rewardHealingMod > 0
+    then
+        totalHealing = totalHealing + math.floor(totalHealing * rewardHealingMod / 100)
+    end
+
+    local diff = petMaxHP - petCurrentHP
+
+    if diff < totalHealing then
+        totalHealing = diff
+    end
+
+    pet:addHP(totalHealing)
+    pet:wakeUp()
+
+    -- Apply regen xi.effect.
+
+    pet:delStatusEffect(xi.effect.REGEN)
+    pet:addStatusEffect(xi.effect.REGEN, regenAmount, 3, regenTime) -- 3 = tick, each 3 seconds.
+    player:removeAmmo()
+
+    pet:updateEnmityFromCure(pet, totalHealing)
+
+    return totalHealing
+end
+
+-- On Ability Check Unleash
+xi.job_utils.beastmaster.onAbilityCheckUnleash = function(player, target, ability)
+    ability:setRecast(math.max(0, ability:getRecast() - player:getMod(xi.mod.ONE_HOUR_RECAST) * 60))
+
+    return 0, 0
+end
+
+-- On Ability Use Unleash
+xi.job_utils.beastmaster.onUseAbilityUnleash = function(player, target, ability)
+    player:addStatusEffect(xi.effect.UNLEASH, 9, 0, 60)
+end
+
+-- On Ability Check For Leave, Heel and Stay.
+xi.job_utils.beastmaster.onAbilityCheckNilPet = function(player, target, ability)
```

Comment:
This function needs to check if your pet is a jug pet (or charmed pet). Leave/Heel/Stay shouldn't work on drg wyverns, automatons, summoner avatars, etc.

Reply:
It looks like other functions need that check too, might need a small refactor.

Reply:
I think I understand what you are saying, for this PR, I didn't change anything that was already in current code base *except* for the level sync change. Not sure if that is in the scope of this PR?

Reply:
BUT, would it be something like this?

`        if player:isJugPet() and
        pet:getObjType() ~= xi.objType.MOB`

Reply:
although arent all the commands different for each like leave/release/dismiss/deactivate all from main Job?

Reply:
For generic ability check for "can use ability"
```
if
    player:isJugPet() or
    pet:getObjType() == xi.objType.MOB
then
```
I think the one you pasted is from familiar, which does behave differently for jug vs non-jug(?)

Reply:
p.s. I hate "isJugPet", but thats not our problem here :)

Reply:
> although arent all the commands different for each like leave/release/dismiss/deactivate all from main Job?

They are, but for some reason there's conditions where they show up when they should not. I recently saw Dismiss as DRG/PUP or PUP/DRG with an automation out. It's just a safety thing until I can find time to fix that

Reply:
Interesting - i just went on a bst/pup and didnt see it in menu - not saying its not there but

Reply:
It's pure paranoia, not gonna lie
![absolutely not me when I see a SMN/BST](https://upload.wikimedia.org/wikipedia/commons/thumb/3/37/Manwithtinfoilhat.jpg/800px-Manwithtinfoilhat.jpg)

Reply:
haha love it!